pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeapiserver
+
+import (
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+
+	restarterapi "github.com/gardener/dependency-watchdog/pkg/restarter/api"
+	scalerapi "github.com/gardener/dependency-watchdog/pkg/scaler/api"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	// DependencyWatchdogExternalProbeSecretName is the name of the kubecfg secret with internal DNS for external access.
+	DependencyWatchdogExternalProbeSecretName = "dependency-watchdog-external-probe"
+	// DependencyWatchdogInternalProbeSecretName is the name of the kubecfg secret with cluster IP access.
+	DependencyWatchdogInternalProbeSecretName = "dependency-watchdog-internal-probe"
+)
+
+// DependencyWatchdogEndpointConfiguration returns the configuration for the dependency watchdog (endpoint role)
+// ensuring that its dependant pods are restarted as soon as it recovers from a crash loop.
+func DependencyWatchdogEndpointConfiguration() (map[string]restarterapi.Service, error) {
+	return map[string]restarterapi.Service{
+		v1beta1constants.DeploymentNameKubeAPIServer: {
+			Dependants: []restarterapi.DependantPods{
+				{
+					Name: v1beta1constants.GardenRoleControlPlane,
+					Selector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      v1beta1constants.GardenRole,
+								Operator: metav1.LabelSelectorOpIn,
+								Values:   []string{v1beta1constants.GardenRoleControlPlane},
+							},
+							{
+								Key:      v1beta1constants.LabelRole,
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{v1beta1constants.ETCDRoleMain},
+							},
+							{
+								Key:      v1beta1constants.LabelRole,
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{v1beta1constants.LabelAPIServer},
+							},
```

Comment:
What do you think about:
```suggestion
								Values:   []string{v1beta1constants.ETCDRoleMain, v1beta1constants.LabelAPIServer},
							},
```

It was not configured like this previously, but should have the same effect.

Reply:
Yes, it should. @amshuman-kr any comments? Why wasn't it done that way right away?

Reply:
@vpnachev's suggestion is equivalent and more compact. I am not sure why it was done using two separate expression entries before.
I will check the rest of the PR later today. Sorry for the delay.